File: tasking/sources/tasking.8.c
/*
* @@name:	tasking.8
* @@type:	C
* @@operation:	compile
* @@expect:	success
* @@version:	omp_3.0
*/
int tp;
#pragma omp threadprivate(tp)
int var;
void work()
{
#pragma omp parallel
    {
        /* do work here */
#pragma omp task
        {
            tp++;
            /* do work here */
#pragma omp task
            {
                /* do work here but don't modify tp */
            }
            var = tp; //Value does not change after write above
        }
    }
}

File: tasking/sources/tasking.8.f
! @@name:	tasking.8
! @@type:	F-fixed
! @@operation:	compile
! @@expect:	success
! @@version:	omp_3.0
      module example
      integer tp
!$omp threadprivate(tp)
      integer var
      contains
      subroutine work
!$omp parallel
         ! do work here
!$omp task
         tp = tp + 1
         ! do work here
!$omp task
           ! do work here but don't modify tp
!$omp end task
         var = tp    ! value does not change after write above
!$omp end task
!$omp end parallel
      end subroutine
      end module
